index.bs

```diff
@@ -133,50 +133,93 @@ run the following steps:
 
 ## Permissions ## {#permissions}
 
-Issue: Figure out some way to integrate this better with the permissions API [[permissions]],
-for example by making the entry be a part of the PermissionDescriptor.
+The <dfn for=PermissionName enum-value>"file-system"</dfn> [=powerful feature=]'s
+permission-related algorithms and types are defined as follows:
+
+: [=permission descriptor type=]
+:: {{FileSystemPermissionDescriptor}}, defined as:
+  <xmp class=idl>
+  enum FileSystemPermissionMode {
+    "read",
```

Comment:
IndexedDB uses "readonly" and "readwrite". 

Do you think it'd be worth aligning? You'd then add "writeonly" later.

Reply:
I initially started out with readonly. I switched to "read" since it seemed weird if query-ing for "readonly" access returns true when you have read and write access.

Maybe the we should have just "read" and "write", and specify the "mode" as a sequence in the permission descriptor. But then we'd need to do something if somebody is just requesting write access. Always requesting "read" and "write" if the website just asks for "write" would work, but makes it harder to later add true write-only access without breaking existing websites. I guess the other option would be to reject if you ask for just "write" access but that seems weird too, especially if a website already has read access.

So given that we're not doing write-only yet, this seemed the least bad of all the options. WDYT?

Reply:
We discussed this off-thread. Current approach SGTM, with the understanding that we'll ask for TAG's opinion during review.